baseplate/experiments/providers/variant_sets/rollout_variant_set.py

```diff
@@ -0,0 +1,56 @@
+from .base import VariantSet
+
+
+class RolloutVariantSet(VariantSet):
+    """VariantSet designed for feature rollouts. Takes a single variant.
+
+    Changing the size of the variant will minimize the treatment of bucketed
+    users. Those users going from no treatment to the provided treatment
+    (or vice versa) are limited to the change in the provided treatment size.
+    For instance, going from 45% to 55% will result in only the new 10% of
+    users changing treatments. The initial 45% will not change. Conversely,
+    going from 55% to 45% will result in only 10% of users losing the
+    treatment.
+    """
+
+    def __init__(self, variants, num_buckets=1000):
+        """ :param variants -- array of dicts, each containing the keys 'name'
+            and 'size'. Name is the variant name, and size is the fraction of
+            users to bucket into the corresponding variant. Sizes are expressed
+            as a floating point value between 0 and 1.
+        :param num_buckets -- the number of potential buckets that can be
+            passed in for a variant call. Defaults to 1000, which means maximum
+            granularity of 0.1% for bucketing
+        """
+        self._validate_variants(variants)
+        self.variant = variants[0]
+        self.num_buckets = num_buckets
+
+    def __contains__(self, item):
+        return self.variant.get('name') == item
+
+    def _validate_variants(self, variants):
+
+        if variants is None:
+            raise ValueError('No variants provided')
+
+        if len(variants) != 1:
+            raise ValueError("Rollout variant only supports one variant.")
+
+        size = variants[0].get('size')
+        if size is None or size < 0.0 or size > 1.0:
```

Comment:
Since you're not doing math, the floating point stuff shouldn't come up but it still might be better to do the same `int(size * num_buckets)` trick here as well.

Reply:
Didn't change this since it's not necessary, and I want to do checks as close to the provided config as possible.